src/Microsoft.DotNet.Interactive/JavaScriptKernel.cs

```diff
@@ -22,5 +22,17 @@ public JavaScriptKernel() : base(DefaultKernelName)
         {
             return FrontendEnvironment.ExecuteClientScript(command.Code, context);
         }
+
+        internal override async Task HandleAsync(KernelCommand command, KernelInvocationContext context)
+        {
+            TrySetHandler(command, context);
+
+            if (CommandTypeIsRegistered(command.GetType()) && command.Handler is null)
+            {
+                command.Handler = (kernelCommand, invocationContext) => FrontendEnvironment.ForwardCommand(kernelCommand, invocationContext);
```

Comment:
This means that the command will have its handler assigned twice. This is kind of confusing. This kernel should have the real handler and the implementation of that real handler should transparently forward the command.

Reply:
the handler will only be assigned once, if we end in the if statement is because this is a command we want to forward to the client side kernel

Reply:
I mean we have two different blocks of code responsible for it, one in `TrySetHandler` and another here. This can be more concise and explicit by letting the handlers in this kernel forward directly. Remember not all commands have settable handlers, so this code won't work for some commands.

Reply:
I have changed that part so that at registration time the jskernel creates default handlers to forward commands to clietnside kernel.  This way we only have 1 logic for handler assignment